pyfarm/agent/service.py

```diff
@@ -118,6 +115,60 @@ def agents_endpoint(cls):
         """
         return config["master_api"] + "/agents/"
 
+    @inlineCallbacks
+    def schedule_call(self, delay, function, *args, **kwargs):
+        """
+        Schedules ``function`` to be run after ``delay`` has elapsed.
+
+        :param int delay:
+            Number of seconds to delay calling ``function``
+
+        :param function:
+            A callable function to run
+
+        :param args:
+            Arguments to pass into ``function``
+
+        :param kwargs:
+            Keywords to pass into ``function``
+
+        :keyword bool now:
+            If True then run ``function`` right now in addition
+            to scheduling it.
+
+        :keyword int repeat:
+            Repeat calling ``function`` this may times.  If not provided
+            then we'll continue to repeat calling ``function`` until
+            the agent shuts down.
+        """
+        if self.shutting_down:
+            svclog.debug(
+                "Skipping task %s(*%r, **%r) [shutting down]",
+                function.__name__, args, kwargs
+            )
+            returnValue(None)
+
+        now = kwargs.pop("now", True)
+        repeat = kwargs.pop("repeat", None)
+        assert isinstance(delay, NUMERIC_TYPES[:-1])
+        assert callable(function)
+        assert repeat is None or isinstance(repeat, int)
+
+        if now:
+            svclog.debug(
+                "Executing task %s(*%r, **%r).  Next execution in %s seconds.",
+                function.__name__, args, kwargs, delay
+            )
+            yield deferLater(self.reactor, 0, function, *args, **kwargs)
+
+        if repeat is None or repeat > 0:
+            repeat -= 1
+            kwargs.update(repeat=repeat)
+            yield deferLater(
+                self.reactor, delay, self.schedule_call, delay, function,
+                *args, **kwargs
+            )
```

Comment:
I'll add a note in the doc string about this.  I agree that there could be a decent amount of drift here due to how the operates but I also agree that it's not the goal to deal in absolute times here either.  But, that should be more clear in the documentation string.